src/daemon/rpmostreed-transaction-types.cxx

```diff
@@ -1274,25 +1274,48 @@ deploy_transaction_execute (RpmostreedTransaction *transaction,
           return glnx_throw (error, "No overrides for package '%s'", name_or_nevra);
         }
 
-      changed = TRUE;
-    }
-
-  if (self->override_replace_local_pkgs)
-    {
-      g_autoptr(GPtrArray) pkgs = NULL;
-      if (!import_many_local_rpms (repo, self->override_replace_local_pkgs, &pkgs,
-                                   cancellable, error))
-        return FALSE;
-
-      if (pkgs->len > 0)
+      if (self->override_replace_local_pkgs)
         {
-          g_ptr_array_add (pkgs, NULL);
-          if (!rpmostree_origin_add_overrides (origin, (char**)pkgs->pdata,
-                                               RPMOSTREE_ORIGIN_OVERRIDE_REPLACE_LOCAL,
-                                               error))
+          g_autoptr(GPtrArray) pkgs = NULL;
+          if (!import_many_local_rpms (repo, self->override_replace_local_pkgs, &pkgs,
+                                       cancellable, error))
             return FALSE;
-          changed = TRUE;
-        }
+
+          for (char **it = (char**)pkgs->pdata; it && *it; it++)
```

Comment:
The `pkgs->pdata` isn't `NULL`-terminated here so that will cause a segfault when it reaches the end of the array. You can either move the `g_ptr_array_add (pkgs, NULL);` up, or use e.g. `for (guint i = 0; i < pkgs->len; i++)` instead (slight preference for the latter, since it's nice to have the array be empty if there's no packages).